### PR TITLE
@/RoboShield fix misc issues

### PR DESCRIPTION
**Pull request by m453h**
## Description

This PR fixes several errors in the RoboShield application namely:
1. Warning(s) logged in console due to incorrect props/parameters passed in `TwoToneBackground` and `Footer` components as shown below:
Screenshot 2024-07-16 at 18 42 45

2. Failing `build-payload` and `generate:types` due to missing `PAYLOAD_PUBLIC_APP_URL` variable

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
👍🏽 @m453h ... love the UI fixes.

Can you add a bit of context on the Payload/TypeScript fixes e.g. both `payload.config.ts` and `server.t`s loads `.env` (and `.env.local`) file so they should be able to pick up `PAYLOAD_PUBLIC_APP_URL` env var from there (I'm assuming `payload build` and `payload generate:types` loads one or both of these entry files).

**Review by koechkevin — APPROVED** [review]
@m453h You'll need to bump version in `package.json` to trigger production deployment

**Comment by m453h** [comment]
> 👍🏽 @m453h ... love the UI fixes.
> 
> Can you add a bit of context on the Payload/TypeScript fixes e.g. both `payload.config.ts` and `server.t`s loads `.env` (and `.env.local`) file so they should be able to pick up `PAYLOAD_PUBLIC_APP_URL` env var from there (I'm assuming `payload build` and `payload generate:types` loads one or both of these entry files).

Yes they both load the `.env `and `.env.local,` however after digging around I observed that when the `payload.config.ts` loads the file(s) it fails to resolve  `PAYLOAD_PUBLIC_APP_URL`,  as we have it set to `PAYLOAD_PUBLIC_APP_URL=$NEXT_PUBLIC_APP_URL`

Two alternative approaches I thought of are :
1. To explicitly set the default URL value in the .env file
2. To discard `dotnev` and use  `loadEnvConfig` in  `payload.config.ts` to load the `.env` file(s)

**Review by kelvinkipruto — APPROVED** [review]

**Comment by kilemensi** [comment]
Nice!!! Yes, `$NEXT_PUBLIC_APP_URL` is a Next.js extensions.

Cool, my suggestions would be:

1. Try `loadEnvConfig` in `payload.config.ts`. If it works, lets use that approach.
2. If it doesn't work, lets explicitly set default value to `PAYLOAD_PUBLIC_APP_URL` but also update docs to reflect that it's now required.

**Comment by m453h** [comment]
> Nice!!! Yes, `$NEXT_PUBLIC_APP_URL` is a Next.js extensions.
> 
> Cool, my suggestions would be:
> 
> 1. Try `loadEnvConfig` in `payload.config.ts`. If it works, lets use that approach.
> 2. If it doesn't work, lets explicitly set default value to `PAYLOAD_PUBLIC_APP_URL` but also update docs to reflect that it's now required.

Cool, option 1. with `loadEnvConfig` works.

File: apps/roboshield/payload.config.ts [hunk 1]
```diff
@@ -122,4 +122,7 @@ export default buildConfig({
     }),
   ] as any[],
   telemetry: process?.env?.NODE_ENV !== "production",
+  typescript: {
+    declare: false,
+  },
```
Comment (kilemensi) [inline apps/roboshield/payload.config.ts:127]: What are the benefits of this?

**Review by kilemensi — APPROVED** [review]

File: apps/roboshield/payload.config.ts (hunk 1 above)
Comment (m453h) [reply apps/roboshield/payload.config.ts:127]: By default its set to true, so when we generate payload types at the end of the generated `payload-types.ts` we will get a module and an exported interface. 
```
declare module 'payload' {
  export interface GeneratedTypes extends Config {}
}
```

However when I do that , I end up getting a name collision on my end. 

Screenshot 2024-07-17 at 10 24 56

Comment (kilemensi) [reply apps/roboshield/payload.config.ts:127]: Good catch... it's a collision between `GeneratedTypes` from payload and what?

Comment (m453h) [reply apps/roboshield/payload.config.ts:127]: Hmm....the auto `GeneratedTypes` are colliding with whats in  `node_modules/payload/dist/index.d.ts`

Comment (m453h) [reply apps/roboshield/payload.config.ts]: I'll need to do further investigation on this will merge it, to main as it is

**Pull request merged**